sphinx/tutorials/SquareMembrane.m

```diff
@@ -0,0 +1,243 @@
+%
+% Accordingly, we can __rearrange__ the sequence of the __same basis vectors__, in order to make the invariance of the two 2-dimensional subspaces contained in that isotypic component evident. For this reason, we define a new basis from the one that *RepLAB* provides by interchanging the 7th and 8th column:
+%
+
+basis_new = [basis(:,1:6) basis(:,8) basis(:,7) basis(:,9)];
+
+% Relative to the rearranged sequence of the same basis vectors, $M$ has the structure:
+
+M_block_new = inv(basis_new)*M*basis_new
+
+% As a result, having explicitly made use of the symmetries of our problem, we have reduced the eigenvalue problem of M to the $c_j$-dimensional eigenvalue problems:
+%
+% \begin{align*}
+% |M_j - \lambda \mathbb{I}_{c_j}| = 0 \quad j=1,2,\dots, N
+% \end{align*}
+%
+% where $M_j$ are the blocks of length $c_j$ that constitute the matrix $M$ and indicate the invariant nature of the subspaces.
+%
+% In other words, instead of solving the full eigenvalue problem for the sparse matrix $M$, we only have to take into account the $N$ blocks and solve a $c_j$-dimensional eigenvalue in each block. In this case, it can be checked that the characteristic polynomial of $M$ factors into a cubic and a quadratic equation.
+
+% ## Finding the eigenvalues and the eigenspaces
+
+% We are now ready to solve the eigenvalue problem for the block-diagonal matrix M and compute the eigenspaces that correspond to the eigenvalues $\lambda$. In order to be clear which eigenstate corresponds to which eigenvalue, we sort the eigenvalues returned from the *MATLAB/Octave* **eig** function from the lowest to the highest and then we rearrange the columns in this order. This way, it is apparent that the first eigenvalue is the upper left diagonal element of the $E_{new}$ matrix, whereas the associated eigenvector is the first column of the $V_{new}$ matrix, as exemplified below:
+
+[V,E] = eig(M_block); % Solve the eigenvalue problem
+E_new = diag(sort(diag(E), 'ascend')) % make diagonal matrix out of sorted diagonal values of input E
+[c,ind]=sort(diag(E),'ascend'); % store the indices of which columns the sorted eigenvalues come from
```

Comment:
Well, I have reformulated this part and used the blocks in each isotypic component to avoid using the 9x9 matrices all over again. 

It seems that the ``eig`` function in Octave does not return the eigenvalues in a particular standard order; I get either ascending or descending, so I need to sort out the eigenvalues again to avoid interchanging of the ground and highest states (since I am plotting this manually in this tutorial).

Is it still better to consider each block seperately? I mean, it is the same procedure twice. We could use this type of sorting: 

``[V,E] = eig(M_block); ``
``[rows,cols]=sort(eig(M_block)); ``
``E_new = diag(rows)`` 
``V_new = V(:,cols) ``

then just display the eigenvalues as `rows`, not the matrix `V` with the coefficients, and explain how to get the eigenspaces for only the ground and the highest state when we make the plots.